test/unit/eval_algorithms/test_qa_accuracy_semantic_robustness.py

```diff
@@ -269,423 +205,77 @@ def test_qa_accuracy_semantic_robustness_evaluate_sample(self, test_case):
-            (test_case.perturbed_model_output_2,),
-        ]
-
-        eval_algorithm = QAAccuracySemanticRobustness(test_case.config)
-        assert (
-            eval_algorithm.evaluate_sample(
-                model_input=test_case.model_input,
-                model=model,
-                target_output=test_case.target_output,
-                model_output=test_case.original_model_output,
-            )
-            == test_case.expected_response
-        )
-        assert model.predict.call_count == 2
-
-    class TestCaseQAAccuracySemanticRobustnessEvaluateSampleInvalid(NamedTuple):
-        model_input: str
-        target_output: str
-        model: ModelRunner
-        expected_error_message: List[EvalScore]
-
-    @pytest.mark.parametrize(
-        "test_case",
-        [
-            TestCaseQAAccuracySemanticRobustnessEvaluateSampleInvalid(
-                model_input="What is the capital of England?",
-                target_output="London",
-                model=None,
-                expected_error_message="Missing required input: model i.e. ModelRunner, for QAAccuracySemanticRobustness "
-                "evaluate_sample",
-            ),
-            TestCaseQAAccuracySemanticRobustnessEvaluateSampleInvalid(
-                model_input=None,
-                target_output="London",
-                model=MagicMock(),
-                expected_error_message="Missing required input: model_input, for QAAccuracySemanticRobustness "
-                "evaluate_sample",
-            ),
-            TestCaseQAAccuracySemanticRobustnessEvaluateSampleInvalid(
-                model_input="What is the capital of England?",
-                target_output=None,
-                model=MagicMock(),
-                expected_error_message="Missing required input: target_output, for QAAccuracySemanticRobustness "
-                "evaluate_sample",
-            ),
-        ],
-    )
-    def test_qa_accuracy_semantic_robustness_evaluate_sample_invalid_input(self, test_case, config):
-        """
-        GIVEN invalid inputs
-        WHEN QAAccuracySemanticRobustness.evaluate_sample is called
-        THEN correct exception with proper message is raised
-        """
-        eval_algorithm = QAAccuracySemanticRobustness(config)
-        with pytest.raises(EvalAlgorithmClientError, match=test_case.expected_error_message):
-            eval_algorithm.evaluate_sample(test_case.model_input, test_case.model, test_case.target_output)
-
-    class TestCaseQAAccuracySemanticRobustnessEvaluate(NamedTuple):
-        input_dataset: Dataset
-        input_dataset_with_generated_model_output: Dataset
-        prompt_template: Optional[str]
-        dataset_config: Optional[DataConfig]
-        expected_response: List[EvalOutput]
-        save_data: bool
+    class TestCaseEvaluate(NamedTuple):
+        user_provided_prompt_template: Optional[str]
+        dataset_prompt_template: str
 
     @pytest.mark.parametrize(
```

Comment:
These are all test cases that don't actually validate the numerical values, but rather ensure that correct function calls are made. `evaluate_dataset` now handles all of this logic, so we can get rid of these test cases. Notice how all of scores are just 0.0, since we're mocking everything.